### PR TITLE
Use json library

### LINKED ISSUE #125: Use a library to write JSON
With the increasing number of places where we generate JSON, it would be good to start using a proper library to do it: https://github.com/nlohmann/json

Where is JSON already used:
- changes
- summary of changes
- conflicts
- database schema

### PR DESCRIPTION
**Pull request by alexbruy**
Integrate https://github.com/nlohmann/json into geodiff and use it to write JSON for:

- changes
- summary of changes
- conflicts
- database schema

Fixes #125 
Fixes https://github.com/lutraconsulting/geodiff/issues/166

### CODE REVIEW

File: geodiff/src/geodiff.cpp [hunk 1]
```diff
@@ -29,6 +29,11 @@
 #include <string>
 #include <vector>
 
+#include "json.hpp"
+
+using json = nlohmann::json;
```
Comment (PeterPetrik) [inline geodiff/src/geodiff.cpp:34]: I am not big fan of `using ...` since for example then you cannott have local variable names called "json" and use use `j` instead. ...

but if @wonder-sk is happy I am easy :)

File: geodiff/tests/testdata/2_deletes/base-deleted_A.json [hunk 2]
```diff
@@ -1,26 +1,26 @@
 {
```
Comment (PeterPetrik) [inline geodiff/tests/testdata/2_deletes/base-deleted_A.json:1]: why do we need to change json (reference) files in the tests? it will be better to compare JSON-s in test regardless of the order of elements? or is it possible to produces same output JSON files as before?

**Review by PeterPetrik — APPROVED** [review]

File: geodiff/tests/testdata/complex/complex1.json [hunk 3]
```diff
@@ -58,7 +58,7 @@
           },
           {
               "column": 2,
-              "new": "\n\t\r\"';,"
+              "new": "\\n\\t\\r\"';,"
```
Comment (PeterPetrik) [inline geodiff/tests/testdata/complex/complex1.json:61]: why this changes? was it buggy before?

Comment (alexbruy) [reply geodiff/tests/testdata/complex/complex1.json:61]: Caret return and some other characters should be escaped. So yes, previously if string had such characters the resulting JSON was incorrect. That's why we got https://github.com/lutraconsulting/qgis-mergin-plugin/issues/378 and added workaround to the QGIS plugin.

Comment (alexbruy) [reply geodiff/tests/testdata/complex/complex1.json:61]: I think yes. We get it for free just be switching to that JSON library, seems it does all necessary escaping and quoting for us.

Comment (PeterPetrik) [reply geodiff/tests/testdata/complex/complex1.json:61]: nice
